cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java

```diff
@@ -506,18 +506,42 @@ protected HttpResponse addAppArtifact(Id.Artifact artifactId, Class<?> cls) thro
   protected HttpResponse addPluginArtifact(Id.Artifact artifactId, Class<?> cls,
                                            Manifest manifest,
                                            Set<ArtifactRange> parents) throws Exception {
+    return addPluginArtifact(artifactId, cls, manifest, parents, null);
+  }
 
+  /**
+   * adds a plugin artifact. This method is present for testing invalid plugin class json and it usage is not
+   * recommended.
+   * @param artifactId the artifact id
+   * @param cls the application class
+   * @param manifest manifest
+   * @param parents parents range of artifact
+   * @param pluginClassesJson JSON array represetation of a collection of plugin class
+   * @return {@link HttpResponse} recieved response
+   * @throws Exception
+   */
+  protected HttpResponse addPluginArtifact(Id.Artifact artifactId, Class<?> cls,
+                                           Manifest manifest,
+                                           @Nullable Set<ArtifactRange> parents,
+                                           @Nullable  String pluginClassesJson) throws Exception {
     Location appJar = PluginJarHelper.createPluginJar(locationFactory, manifest, cls);
     try {
-      return addArtifact(artifactId, Locations.newInputSupplier(appJar), parents);
+      return addArtifact(artifactId, Locations.newInputSupplier(appJar), parents, pluginClassesJson);
     } finally {
       appJar.delete();
     }
   }
 
   // add an artifact and return the response code
   protected HttpResponse addArtifact(Id.Artifact artifactId, InputSupplier<? extends InputStream> artifactContents,
-                                     Set<ArtifactRange> parents) throws Exception {
+                                     @Nullable Set<ArtifactRange> parents) throws Exception {
```

Comment:
we should avoid nullable collections and force callers to pass in an empty set instead of a null